### DISCUSSION TITLE
How to get response headers from the stream API?

### DISCUSSION THREAD

**Discussion by SevenOutman** [sindresorhus/got#1867] [Q&A] [answered]
I need to read the response headers when I download files with `got.stream()`. But there seems to be no explanation how this should be done.

**Comment by sindresorhus** [comment, marked as answer]
https://github.com/sindresorhus/got/tree/f896aa52abc41fe40d4942da94a0408477358f14#onresponse-response

**Reply by SevenOutman** [reply]
Thanks for your answer!

I've got a further question - is there any events that fires right after headers have been received, so that I can get the headers earlier (before the body has been received)?

**Reply by surjyams** [reply]
It would be extremely helpful if we could get the response data (the stream) and headers accessible separately.

**Reply by leafac** [reply]
I think I need this as well. I’m trying to allowlist headers before I pipe the stream down to an HTTP response, and so far the only way I think I could do that was to buffer the whole Got response, and streaming would of course be better.
